samples/jvm/src/main/java/guru/stefma/cleancomponents/usecase/sample/jvm/Sample.kt

```diff
@@ -43,44 +46,45 @@ class GetUserUseCase(
  */
 @UseCase
 class GetUsePointsForUserIdUseCase(
-        private val getUser: GetUser,
-        override val executionScheduler: Scheduler,
-        override val postExecutionScheduler: Scheduler
+    private val getUser: GetUser,
+    override val executionScheduler: Scheduler,
+    override val postExecutionScheduler: Scheduler
 ) : SingleUseCase<Int, GetUsePointsForUserIdUseCase.Params> {
 
     override fun buildUseCase(params: GetUsePointsForUserIdUseCase.Params): Single<Int> {
         return getUser.buildUseCase(
-                guru.stefma.cleancomponents.usecase.sample.jvm.GetUserUseCase.Params(params.userId))
-                .flatMap {
-                    Single.just(it.authorizationKey)
-                            .map { 99 }
-                }
+            GetUserUseCase.Params(params.userId)
+        ).flatMap {
+            Single.just(it.authorizationKey)
+                .map { 99 }
+        }
     }
 
     data class Params(val userId: String)
 }
 
 @UseCase
-class GetUserCoroutineUseCase : CoroutineUseCase<User, GetUserCoroutineUseCase.Params> {
+class GetUserCoroutineUseCase(
+    override val callbackDispatcher: CoroutineContext = Dispatchers.IO
+) : CoroutineUseCase<User, GetUserCoroutineUseCase.Params>() {
 
-    override suspend fun execute(params: Params): User {
-        return User(params.userId, "Thorsten", Gender.MALE, Random().nextInt(1000).toString())
+    override suspend fun execute(params: Params): Result<User> {
+        return Result.success(User(params.userId, "Thorsten", MALE, Random().nextInt(1000).toString()))
     }
 
     data class Params(val userId: String)
-
 }
 
 @UseCase
-class GetUsePointsForUserIdCoroutineUseCase(private val getUser: GetUserCoroutine) : CoroutineUseCase<Int, GetUsePointsForUserIdCoroutineUseCase.Params> {
-
-    override suspend fun execute(params: Params): Int {
-        val user = getUser.execute(GetUserCoroutineUseCase.Params(params.userId))
-        return user.authorizationKey.run {
-            99
-        }
+class GetUsePointsForUserIdCoroutineUseCase(
+    private val getUser: GetUserCoroutine,
+    override val callbackDispatcher: CoroutineContext = Dispatchers.IO
+) : CoroutineUseCase<Int, GetUsePointsForUserIdCoroutineUseCase.Params>() {
+
+    override suspend fun execute(params: Params): Result<Int> {
+        val user = getUser.execute(GetUserCoroutineUseCase.Params(params.userId)).getOrThrow()
+        return Result.success(user.authorizationKey.run { 99 })
```

Comment:
The getUser should do a map somehow, no? 🤔

Reply:
Map to what?

Reply:
This implementation is a little bit different to the
`GetUsePointsForUserIdUseCase`.
It should be "normally":
```
Result.success(user. authorizationKey)
    .map(99)
```
That would be a 1:1 copy of the RxJava sample...

Can you please change this 🙏

Reply:
This doesn't work, because it would map a `Result` to an `Int` and then it has a wrong return type. Also the `run` cannot be a `map` because the return type then is `List<Int>`